### ISSUE TITLE
Please add the functionality to save tokenizer model for run_language_modeling.py

### ISSUE THREAD

**Issue by vincentwen1995** [huggingface/transformers#4695]
# 🚀 Feature request

Please add the feature to save the tokenizer model during training to the checkpoint folders.
## Motivation

When I tried out the script for [fine-tuning with language modeling](transformers/examples/language_modeling/run_language_modeling.py), I realized that the generated checkpoints during training cannot allow continue training, because under the checkpoint folders, the corresponding tokenizer model is not saved (including the files: tokenizer_config.json, special_tokens_map.json, vocab.txt). As I checked the [script](https://github.com/huggingface/transformers/blob/0866669e751bef636fa693b704a28c1fea9a17f3/examples/language-modeling/run_language_modeling.py#L250), and noticed that the tokenizer model is only saved after the training process.
## Your contribution

Haven't looked into the code in details so it might be best to have someone familiar with the Trainer class to integrate this.

**Comment by LysandreJik** [comment]
This would be good indeed (cc @julien-c). In the meantime I think you can specify 
```
--tokenizer_name=$TOKENIZER_NAME_OR_PATH
```
 so that it always loads the initial tokenizer (which does not change during training).

**Comment by vincentwen1995** [comment]
> This would be good indeed (cc @julien-c). In the meantime I think you can specify
> 
> ```
> --tokenizer_name=$TOKENIZER_NAME_OR_PATH
> ```
> 
> so that it always loads the initial tokenizer (which does not change during training).

Thank you!

**Comment by EdwardRaff** [comment]
I'm trying to figure out how to use a custom made or create a tokenizer using this script and having significant difficulty. There does not seem to be any documentation on how to do this. I attempted to follow [this ](https://huggingface.co/blog/how-to-train) example but it makes no mention of how the tokenizer gets loaded / used.

**Comment by FerchichiNourchene** [comment]
Hi @EdwardRaff ,
I'm facing the same issue while trying to train BERT model from scratch on my own dateset. Did you figure out how to solve it?
```

OSError: Can't load config for './models/BuckBERTer-small/'. Make sure that:

- './models/BuckBERTer-small/' is a correct model identifier listed on 'https://huggingface.co/models'

- or './models/BuckBERTer-small/' is the correct path to a directory containing a config.json file
```

**Comment by obruker** [comment]
@EdwardRaff @FerchichiNourchene `run_language_modeling.py` doesn't work if a tokenizer is specified, but it does not contain the model configuration files. [This](https://stackoverflow.com/a/64795300/3950710) workaround worked for me.

**Comment by jeromemassot** [comment]
Hi all, 

about the missing config.json in the official EsperBERTO notebook, I solved the issue by adding a simple 

`config.to_json_file("./EsperBERTo/config.json") `

to the notebook. I have noticed that a config.json exists in the checkpoint folder, but the pipeline complains about it.

The pipeline init becomes:

```
fill_mask = pipeline(
    "fill-mask",
    model="./EsperBERTo/checkpoint-10000",
    tokenizer="./EsperBERTo"
)
```

WIth these modifications, the pipeline works and the predicted masked words are consistent.

Best regards

Jerome
